implementations/rust/ockam/ockam_api/src/authenticator/enrollment_tokens/acceptor_worker.rs

```diff
@@ -0,0 +1,71 @@
+use either::Either;
+use minicbor::Decoder;
+use ockam::identity::IdentitySecureChannelLocalInfo;
+use ockam_core::api::{Method, RequestHeader, Response};
+use ockam_core::compat::sync::Arc;
+use ockam_core::{Result, Routed, Worker};
+use ockam_node::Context;
+use tracing::trace;
+
+use crate::authenticator::enrollment_tokens::EnrollmentTokenAcceptor;
+use crate::authenticator::one_time_code::OneTimeCode;
+use crate::authenticator::{
+    secure_channel_required, AuthorityEnrollmentTokenRepository, AuthorityMembersRepository,
+};
+
+pub struct EnrollmentTokenAcceptorWorker {
+    pub(super) acceptor: EnrollmentTokenAcceptor,
+}
+
+impl EnrollmentTokenAcceptorWorker {
+    pub fn new(
+        tokens: Arc<dyn AuthorityEnrollmentTokenRepository>,
+        members: Arc<dyn AuthorityMembersRepository>,
+    ) -> Self {
+        Self {
+            acceptor: EnrollmentTokenAcceptor::new(tokens, members),
+        }
+    }
+}
+
+#[ockam_core::worker]
+impl Worker for EnrollmentTokenAcceptorWorker {
+    type Context = Context;
+    type Message = Vec<u8>;
+
+    async fn handle_message(&mut self, c: &mut Context, m: Routed<Self::Message>) -> Result<()> {
+        let secure_channel_info = match IdentitySecureChannelLocalInfo::find_info(m.local_message())
+        {
+            Ok(secure_channel_info) => secure_channel_info,
+            Err(_e) => {
+                secure_channel_required(c, m).await?;
```

Comment:
One other thing about `secure_channel_required`. I am not a big fan that it is doing a silent reply to the sender.
I would be more comfortable with an overall structure that looks like
```rust
let res = with_secure_channel(m, handle_request(m));
c.send(m.return_route(), res).await
```

Reply:
Makes sense to just remove it altogether